rfc-016-slack-channel-names.md

```diff
@@ -0,0 +1,21 @@
+# Slack channel names
+
+## Summary
+
+Guidelines for slack channel naming conventions building on [Slack's own guidelines](https://get.slack.help/hc/en-us/articles/217626408-Create-guidelines-for-channel-names).
+
+## Problem
+
+There is no standard naming convention for Slack channel names and as we grow they become harder to discover. It's difficult to understand whether a channel is for discussion on topics or with teams. For example some SRE questions go into #live-services, some into #help, #engineering and even into #announcements.
```

Comment:
This feels like two problems rolled into one - I agree with the purposes not being clear, but I've not had trouble finding channels by keywords, even when I've had to reach out to an unrelated team. 

I don't think there are any examples of teams being named something entirely unrelated (e.g. #gov-wifi isn't #we-love-dogs, despite my protests).

I worry that this is a fairly heavy-handed proposal, when we could recommend the use of slack's "channel purpose" headers, and that would serve this purpose.